### ISSUE TITLE
derive.democracy.referendums need outPut the ReferendumIndex

### ISSUE THREAD

**Issue by jiangfuyao** [polkadot-js/api#722]
```javascritp
await api.derive.democracy.referendums((result)=>{
      console.log(result)
    })
```
**echo example:**  
`[{end: 216, proposal: {callIndex: 0x0301, args: {who: 5GfWeNe33QeyzVMV2uENuvSXLVfVXCiuLvfEP6WCDHADZhen, free: 1000000000, reserved: 10000000000}}, threshold: Super majority approval, delay: 600}]`  

Not have `ReferendumIndex`, I think the UI-apps is use the `result.index` to show.  Need to show the chain  ReferendumIndex.  

<img width="1042" alt="2019-02-17 9 14 58" src="https://user-images.githubusercontent.com/34789555/53063758-4119e900-3500-11e9-9d1f-316459ce8a76.png">

**Comment by jiangfuyao** [comment]
> ```
> await api.derive.democracy.referendums((result)=>{
>       console.log(result)
>     })
> ```
> **echo example:**
> `[{end: 216, proposal: {callIndex: 0x0301, args: {who: 5GfWeNe33QeyzVMV2uENuvSXLVfVXCiuLvfEP6WCDHADZhen, free: 1000000000, reserved: 10000000000}}, threshold: Super majority approval, delay: 600}]`
> 
> Not have `ReferendumIndex`, I think the UI-apps is use the `result.index` to show. Need to show the chain ReferendumIndex.
> 
> <img alt="2019-02-17 9 14 58" width="1042" src="https://user-images.githubusercontent.com/34789555/53063758-4119e900-3500-11e9-9d1f-316459ce8a76.png">

**Comment by jiangfuyao** [comment]
oh, UI-apps is use this to echo ReferendumIndex:  
```javascript
const startIndex = referendumCount - democracy_referendums.length;

    return democracy_referendums.map((referendum, index) => (
      <Referendum
        idNumber={index + startIndex}
        key={index}
        value={referendum}
      />
    ));
```

**Comment by jacogr** [comment]
It is actually a display toString issue. The field is there, but not included in the `toString` - fixed in #723 (The index field exists, and actually WIP versions of apps uses exactly that, i.e. https://github.com/polkadot-js/apps/pull/730/files#diff-d32916d991cd9efc2c3f1d25dcb1669cR98)

Just be aware that `0.45.x` aligns with what Substrate provides so some storage entrie will return `Option` - this is one of them. So use would be -

```js
// after the above PR is in, will display `index` as well
api.derive.democracy.referendums((result)=>{
  console.log(result)
})

// use of it with the `Option<Type>` as per https://polkadot.js.org/api/METHODS_STORAGE.html#democracy
api.derive.democracy.referendums((option)=>{
  console.log(result); // works as expected

  const info = option.unwrapOr(null); // just for `Option<...>`

  if (info) {
    console.log(info.index);
  }
})
```
